File: Sorting/Bubble Sort/BubbleSort.c
#include <stdio.h>

// Sorting Function
void sort(int a[], int n) {
    
    for (int i=n-1; i>0; i--) {
        for (int j=1; j<=i; j++) {
            // if (a[j] > a[j-1]) {  // Asending Order
            if (a[j] < a[j-1]) {  // Desending Order
                int temp = a[j];
                a[j] = a[j-1];
                a[j-1] = temp;
            }
        }
    }

}

// DRIVER CODE
int main() {
    // Inputs
    int array[10] = {23, -6, -3, 7, 34, 9, 66, 23, 55, 1};
    int n = 10;
    
    // Unsorted Array Output
    printf("Unsorted Array :-\n");
    for (int i=0; i<n; i++)
        printf("%-4d", array[i]);

    // Sorting Array
    sort(array, n);

    // Sorted Array Output
    printf("\nUnsorted Array :-\n");
    for (int i=0; i<n; i++)
        printf("%-4d", array[i]);

    return 0;
}

File: Sorting/Bubble Sort/BubbleSort.py
def sort(l, n):

    for i in range(n-1, 0, -1):
        for j in range(1, i+1):
            if l[j] < l[j-1]:  # Asending Order
            # if l[j] > l[j-1]:  # Desending Order
                l[j],l[j-1] = l[j-1],l[j]



if __name__ == "__main__":
    # Inputs
    numberList = [23, -6, -3, 7, 34, 9, 66, 23, 55, 1]
    n = len(numberList)

    # Unsorted List Output
    print(numberList)

    # Sorting List
    sort(numberList, n)

    # Sorted List Output
    print(numberList)
